fedmsg_meta_umb/freshmaker.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# fedmsg_meta_umb is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# fedmsg_meta_umb is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with fedmsg; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Authors:  Chenxiong Qi <cqi@redhat.com>
+
+from fedmsg.meta.base import BaseProcessor
+
+
+class FreshmakerProcessor(BaseProcessor):
+    topic_prefix_re = r'/topic/VirtualTopic\.eng'
+
+    __name__ = 'freshmaker'
+    __description__ = 'a service scheduling rebuilds of artifacts as new content becomes available'
+    __link__ = 'https://freshmaker.engineering.redhat.com'
+    __docs__ = 'https://mojo.redhat.com/docs/DOC-1155261'
+    __obj__ = 'Artifact rebuild scheduler'
+
+    def title(self, msg, **config):
+        return msg['topic'].split('.', 2)[-1]
+
+    def subtitle(self, msg, **config):
+        inner_msg = msg['msg']
+        topic = msg['topic']
+
+        if topic.endswith('manual.rebuild'):
+            template = self._('Manual rebuild is triggered for Errata advisory {errata_id}.')
+        elif topic.endswith('build.state.changed'):
+            if inner_msg['state_name'] == 'DONE':
+                template = self._('{type_name} {name} is successfully rebuilt '
+                                  'from original build {original_nvr} to new '
+                                  'build {rebuilt_nvr}.')
+            else:
+                template = self._('{type_name} build state is switched to {state_name} '
+                                  'due to "{state_reason}"')
+        elif topic.endswith('event.state.changed'):
+            template = self._('Event state is switched to {state_name} due to "{state_reason}"')
+        else:
+            template = 'Unknown message.'
+
+        return template.format(**inner_msg)
+
+    def link(self, msg, **config):
+        topic = msg['topic']
+
+        if (topic.endswith('build.state.changed') or
+                topic.endswith('event.state.changed')):
+            return msg['msg']['url']
+        else:
+            # For manual rebuild and other any new kind of message that is
+            # not known yet.
+            return ''
```

Comment:
I recommend returning `None` here.  Callers will expect to not generate `<a href="..."` tags if None is returned.